crmsh/utils.py

```diff
@@ -2287,4 +2253,58 @@ def interface_choice():
     # should consider interface format like "ethx@xxx"
     interface_list = re.findall(r'(?:[0-9]+:) (.*?)(?=: |@.*?: )', out)
     return [nic for nic in interface_list if nic != "lo"]
+
+
+def interface_info(ipv6=False):
+    """
+    Get network information(IPv4 default) in local, include:
+        nic-ip dictionary
+        ip list
+        ip with mask list
+    IPv6 support
+    """
+    nic_ip_dict = {}
+    ip_list = []
+    ip_with_mask_list = []
+
+    cmd = "ip -{} -o addr show".format(6 if ipv6 else 4)
+    rc, out, err = get_stdout_stderr(cmd)
+    if rc != 0:
+        raise ValueError(err)
+
+    for line in out.split('\n'):
+        _, nic, _, ip_with_mask, *_ = line.split()
+        if nic == "lo":
+            continue
+        ip, _ = ip_with_mask.split('/')
+        ip_list.append(ip)
+        ip_with_mask_list.append(ip_with_mask)
+        # get first address on each nic
+        if nic not in nic_ip_dict:
+            nic_ip_dict[nic] = ip
+    return nic_ip_dict, ip_list, ip_with_mask_list
```

Comment:
I don't like this return. It returns a dictionary and 2 list, with really redundant data. Having such a complex return brings a lot of issues in the future and makes the code really difficult to maintain
I would prefer to have a unique dictionary, or even better a class.
Option 1:
```
{
    "eth0" : {
        "ip": "1.2.3.4".
        "mask": 24
    },
# More network interfaces
}
```

Option 2: Create a really simple class that has a list of interfaces, and each interface has the nic, ip address and the mask.